feat: Allow multiple org owners

xrdt opened the pull request:
## Problem

This should reduce the number of support tickets we get asking us to transfer org ownership after the original owner has left.

Changes: 
* Allow multiple org owners 
* Update the UI to allow making other users owners 
* Update the invite UI to allow selecting the level at which you want to invite a new user (defaults to member)

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

Commit:
remove db constraint on multiple org owners

Commit:
update ui to support making other org members owners

Commit:
update invite modal

Commit:
add typing confirmation on invite modal for owner level invites

xrdt commented:
Demo: 

https://github.com/PostHog/posthog/assets/21014901/f2246961-0dcd-46d8-a096-78dcd0209b50

zlwaterfield commented on frontend/src/scenes/settings/organization/Members.tsx:
```diff
@@ -69,11 +69,13 @@ function ActionsComponent(_: any, member: OrganizationMemberType): JSX.Element |
                                             primaryButton: {
                                                 status: 'danger',
-                                                children: 'Transfer Ownership',
+                                                children: `Make ${fullName(member.user)} owner`,
```
Does it sound better as `Make ${fullName(member.user)} an owner`?

xrdt replied:
Yes, good point.

zlwaterfield replied:
Didn't see this get updated

zlwaterfield commented on frontend/src/scenes/settings/organization/InviteModal.tsx:
```diff
@@ -97,18 +112,30 @@ export function InviteRow({ index, isDeletable }: { index: number; isDeletable:
-                )}
             </div>
+            <div className="flex-1 flex gap-1 items-center justify-between">
+                <LemonSelect
```
What happens if a member is inviting someone? Should we just hide this?

xrdt replied:
Will hide if there is only one allowed option.

zlwaterfield reviewed (changes requested):
Awesome work here! Excited to have this :) 

Can we add some tests here since this is such a sensitive area around org ownership.

I'd imagine both stories for visual regress and validation for API would be needed.

Commit:
lint

Commit:
only show level dropdown if there are multiple options

Commit:
some slight modification to backend tests

Commit:
fix migration

github-actions[bot] committed:
Update query snapshots

Commit:
add storybook

xrdt commented:
Thanks for the reviews, @zlwaterfield , have updated existing tests and added a storybook mock.

Commit:
silence some constraint checfk

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

zlwaterfield reviewed (approved):
Just one comment about the button copy. 

Flyby: I think we should add more UI tests here for the modal being open from the perspective of each role type to make sure the dropdown appears as expected but not a blocker, feel free to do that at another time.

Commit:
one more pr feedback

Commit:
slightly spruce up the pending invites table, remove pending invites from modal

Commit:
fix cancel invite modal, adjust right padding on modal headers

Commit:
update stories

xrdt commented:
Some new modifications: 
* Removed the pending invites from the invite modal. You can see and delete these on the page itself so it's not clear that having an additional display for them in the modal is useful. They were also cluttering up the alignment of the modal. 
* Added the level a user is invited as to the pending invites table. 
* Fixed the cancel modal layout. 
* Adjusted LemonModal header right padding so it doesn't overlap with the close button. 
* Added some more storybook mocks to showcase the different options depending on the current user's organization membership level.

Commit:
finally fix the migration

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

zlwaterfield commented on .vscode/launch.json:
```diff
@@ -71,7 +71,7 @@
                 "SKIP_SERVICE_VERSION_REQUIREMENTS": "1",
                 "PRINT_SQL": "1",
-                "BILLING_SERVICE_URL": "https://billing.dev.posthog.dev",
+                "BILLING_SERVICE_URL": "http://localhost:8100", // "https://billing.dev.posthog.dev",
```
Can you put this back?

zlwaterfield commented on frontend/src/scenes/settings/organization/InviteModal.tsx:
```diff
@@ -124,69 +165,11 @@ export function InviteTeamMatesComponent(): JSX.Element {
+                    {allowedLevelsOptions.length > 1 && <b className="flex-1">Level</b>}
                 </div>
 
-                {invitesReversed.map((invite: OrganizationInviteType) => {
```
To confirm this is just removing existing invites from the modal? Works for me.

xrdt replied:
Yeah, I'm removing existing invites. I thought it looked cluttered and didn't think it added anything beyond the pending invites table already on the settings page.

zlwaterfield commented on test.env:
```diff
@@ -0,0 +1,6 @@
+export REDIS_URL='redis://localhost'
```
Remove this file?

Commit:
fix extraneous changes

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
fix frontend lint

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)
